sdk/python/kfp/components/_python_op.py

```diff
@@ -64,37 +64,50 @@ def _capture_function_code_using_cloudpickle(func, modules_to_capture: List[str]
         func_pickle = base64.b64encode(cloudpickle.dumps(func, pickle.DEFAULT_PROTOCOL))
     finally:
         sys.modules.update(old_modules)
-    func_code = '{func_name} = pickle.loads(base64.b64decode({func_pickle}))'.format(
+
+    function_loading_code = '''\
+import sys
+try:
+    import cloudpickle as _cloudpickle
+except ImportError:
+    import subprocess
+    try:
+        print("cloudpickle is not installed. Installing it globally", file=sys.stderr)
+        subprocess.run([sys.executable, "-m", "pip", "install", "cloudpickle==1.1.1", "--quiet"], env={"PIP_DISABLE_PIP_VERSION_CHECK": "1"}, check=True)
```

Comment:
Just a question, since you need to install cloudpickle for lightweight component. Do you think it's a good idea to move most of the lightweight component code into a package. Then, you just need to codegen the entrypoint code and can just use python code to write the rest of it.

Reply:
>Do you think it's a good idea to move most of the lightweight component code into a package. Then, you just need to codegen the entrypoint code and can just use python code to write the rest of it.

I think it might not work in practice as I'll be replacing the code trying to install cloudpickle with the code trying to install that new package. Only the version-checking lines can be moved to that new package.

>need to install cloudpickle for lightweight component

I'm trying to avoid that in long-term by including needed parts in the code (https://github.com/kubeflow/pipelines/issues/1430) or just abandoning fragile pickling in favor of source code based solution.